components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -5212,128 +5160,201 @@ public int compare(LifeCycleEvent o1, LifeCycleEvent o2) {
+                            APIConstants.SubscriptionStatus.ON_HOLD.equals(subStatus) ||
+                            APIConstants.SubscriptionStatus.REJECTED.equals(subStatus)) &&
+                            APIConstants.SubscriptionCreatedStatus.SUBSCRIBE.equals(subCreationStatus)) {
+
+                        //Throw error saying subscription already exists.
+                        log.error(String.format("Subscription already exists for API/API Prouct %s in Application %s"
+                                , apiTypeWrapper.getName(), application.getName()));
+                        throw new SubscriptionAlreadyExistingException(String.format("Subscription already exists for" +
+                                " API/API Prouct %s in Application %s", apiTypeWrapper.getName(), application.getName()));
+
+                    } else if (APIConstants.SubscriptionStatus.UNBLOCKED.equals(subStatus) && APIConstants
+                            .SubscriptionCreatedStatus.UN_SUBSCRIBE.equals(subCreationStatus)) {
+                        deleteSubscriptionByApiIDAndAppID(id, application.getId(), connection);
+                    } else if (APIConstants.SubscriptionStatus.BLOCKED.equals(subStatus) || APIConstants
+                            .SubscriptionStatus.PROD_ONLY_BLOCKED.equals(subStatus)) {
+                        log.error(String.format(String.format("Subscription to API/API Prouct %%s through application" +
+                                " %%s was blocked"), apiTypeWrapper.getName(), application.getName()));
+                        throw new SubscriptionBlockedException(String.format("Subscription to API/API Product %s " +
+                                "through application %s was blocked", apiTypeWrapper.getName(), application.getName()));
+                    } else if (APIConstants.SubscriptionStatus.REJECTED.equals(subStatus)) {
+                        throw new SubscriptionBlockedException("Subscription to API " + apiTypeWrapper.getName()
+                                + " through application " + application.getName() + " was rejected");
                     }
                 }
+
             }
-        } catch (SQLException e) {
-            handleException("Error when executing the SQL queries", e);
+        }
+
+        //This query to update the AM_SUBSCRIPTION table
+        String sqlQuery = SQLConstants.ADD_SUBSCRIPTION_SQL;
+
+        //Adding data to the AM_SUBSCRIPTION table
+        //ps = conn.prepareStatement(sqlQuery, Statement.RETURN_GENERATED_KEYS);
+        PreparedStatement preparedStForInsert = null;
+
+        try {
+            if (connection.getMetaData().getDriverName().contains("PostgreSQL")) {
+                preparedStForInsert = connection.prepareStatement(sqlQuery, new String[]{"subscription_id"});
+            } else {
+                preparedStForInsert = connection.prepareStatement(sqlQuery, new String[]{"SUBSCRIPTION_ID"});
+            }
```

Comment:
This might be better,

```
String subscriptionIDColumn = "SUBSCRIPTION_ID";

 if (connection.getMetaData().getDriverName().contains("PostgreSQL")) {
         subscriptionIDColumn = "subscription_id";
 } 

try (PreparedStatement preparedStForInsert = connection.prepareStatement(sqlQuery, new String[]{subscriptionIDColumn})) {


}
               
```